Update documentation for redis.io (#967)
RedisGears/RedisGears: docs/docs/concepts/JavaScript_API.md

---
title: "JavaScript API"
linkTitle: "JavaScript API"
weight: 2
description: >
    Overview of the JavaScript API
---

## Redis object

* Since version: 2.0.0

> The Triggers and Functions JavaScript API always provides a singleton instance of an object named *redis*. The *redis* instance enables the functions to interact with the Redis server that is running it. Following is the API provided by the *redis* object instance.

Reviewer: ```suggestion
The Triggers and Functions JavaScript API provides a singleton instance of an object named *redis*. The *redis* instance enables registered functions to interact with the Redis server on which they are running. Following is the API provided by the *redis* instance.
```

  onTriggerFired: function(client, data){}
})

redis.registerKeySpaceTrigger(
  'foo', // trigger name
  'keys:*', //key prefix
  function(client, data) {}, //callback
  {
    description: 'description'
    eventTypes: ['hset'],
    keyTypes: ['hash'],
    onTriggerFired: function(client, data){}
  } //optional arguments
)
```

### `redis.registerStreamTrigger`

* Since version: 2.0.0

```JavaScript
redis.registerStreamTrigger({
  name: 'foo',
  description: '',
  prefix: 'stream1',
  window: 1,
  isStreamTrimmed: false,
  callback: function(client, data){}
});

redis.registerStreamTrigger(
  'foo', //trigger name
  'stream:*', //prefix
  function(client, data){},//callback
  {
    description: 'Description'
    window: 1,
    isStreamTrimmed: false 
  } //optional arguments
)
```

### `redis.registerClusterFunction`

* Since version: 2.0.0

```JavaScript
redis.registerClusterFunction(
  'foo', //name
  async function(client, ...args){} //callback
)
```

### `redis.config`

* Since version: 2.0.0

```Shell
redis-cli -x RG.FUNCTION LOAD UPGRADE CONFIG '{"example_config": "example_config_value"}' < main.js
```

```JavaScript
var configExample = redis.config.example_config;
```

> ## Client object

Reviewer 2: We should probably split client to async client and client (async client you get on async function ...)

Author: I added comments that it is only for demo and not to use it in production

Author: Will add this when we expand the docs

Reviewer 2: But this is currently incorrect, you can not do all those operations with both clients

### `client.call`

* Since version: 2.0.0

```JavaScript
client.call(
  '',
  ...args
)
```

### `client.callRaw`

* Since version: 2.0.0

```JavaScript
client.callRaw(
  '',
  ...args
)

> ```

Reviewer 2: You are missing callAsync and callAsyncRaw
